broker.go

```diff
@@ -156,6 +169,8 @@ func (b *Broker) Close() (err error) {
 	b.done = nil
 	b.responses = nil
 
+	atomic.StoreInt32(&b.opened, 0)
+
```

Comment:
We can't *instead of*, but we could do *as well as*, if we wanted to avoid taking the mutex here - I'm not sure that it's all that useful though?